plugins/services/src/js/data/index.ts

```diff
@@ -0,0 +1,164 @@
+import { makeExecutableSchema } from "graphql-tools";
+import { ServicePlanStatusSchema } from "#PLUGINS/services/src/js/types/ServicePlanStatus";
+import { ServicePlanStepSchema } from "#PLUGINS/services/src/js/types/ServicePlanStep";
+import { ServicePlanPhaseSchema } from "#PLUGINS/services/src/js/types/ServicePlanPhase";
+import {
+  ServicePlanDetailResolver,
+  ServicePlanSchema
+} from "#PLUGINS/services/src/js/types/ServicePlan";
+import { Service, ServiceSchema } from "#PLUGINS/services/src/js/types/Service";
+import { Observable } from "rxjs";
+import { RequestResponse } from "@dcos/http-service";
+import {
+  fetchPlanDetails as fetchServicePlanDetail,
+  fetchPlans as fetchServicePlans,
+  ServicePlanResponse
+} from "#PLUGINS/services/src/js/data/ServicePlansClient";
+import Config from "#SRC/js/config/Config";
+
+export interface ResolverArgs {
+  fetchServicePlans: (
+    serviceName: string
+  ) => Observable<RequestResponse<string[]>>;
+  fetchServicePlanDetail: (
+    serviceName: string,
+    planName: string
+  ) => Observable<RequestResponse<ServicePlanResponse>>;
+  pollingInterval: number;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface ServiceQueryArgs {
+  name: string;
+}
+
+function isServiceQueryArgs(args: GeneralArgs): args is ServiceQueryArgs {
+  return (args as ServiceQueryArgs).name !== undefined;
+}
+
+export interface PlansQueryArgs {
+  name: string;
+}
+
+function isPlansQueryArgs(args: GeneralArgs): args is PlansQueryArgs {
+  return (args as PlansQueryArgs).name !== undefined;
+}
+
+function handleAPIError(reqResp: RequestResponse<any>, message: string = "") {
+  if (reqResp.code !== 200) {
```

Comment:
Should we do the same check as in "ServicePlansClient" here - https://github.com/dcos/dcos-ui/pull/3503/files#diff-709253ad12d4f5ba0075cd28d643b130R41 ?